pkg/packagekit/history.jsx

```diff
@@ -29,22 +29,34 @@ import cockpit from "cockpit";
 
 const _ = cockpit.gettext;
 
-function formatPkgs(pkgs) {
-    const names = Object.keys(pkgs).filter(i => i != "_time");
-    names.sort();
-    return names.map(n => {
-        const tooltipRef = React.useRef(null);
+const PackageItem = ({ name, value }) => {
+    const tooltipRef = React.useRef(null);
+
+    return (
+        <React.Fragment>
+            <li ref={tooltipRef}>{name}</li>
+            <Tooltip triggerRef={tooltipRef} content={`${name} ${value}`} />
+        </React.Fragment>
+    );
+};
 
-        return (
-            <React.Fragment key={n}>
-                <li ref={tooltipRef}>{n}</li>
-                <Tooltip triggerRef={tooltipRef} content={ n + " " + pkgs[n] } />
-            </React.Fragment>
-        );
-    });
-}
+export const PackageList = ({ packages }) => {
+    if (!packages) {
+        return null;
+    }
 
-export const PackageList = ({ packages }) => packages ? <ul className='flow-list'>{formatPkgs(packages)}</ul> : null;
+    const packageNames = Object.keys(packages)
+        .filter(name => name !== "_time")
+        .sort();
```

Comment:
[`sort()` is done in-place](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) so this wouldn't work. That's why it was split in the `formatPkgs()` function too.
```suggestion
    const packageNames = Object.keys(packages)
        .filter(name => name !== "_time");
    packageNames.sort();
```